filecoin-proofs/src/api/post_adapter.rs

```diff
@@ -0,0 +1,626 @@
+use crate::api::internal::ChallengeSeed;
+use crate::api::internal::Commitment;
+use crate::api::internal::POST_SECTORS_COUNT;
+use crate::error;
+use sector_base::api::bytes_amount::PaddedBytesAmount;
+use std::iter::repeat;
+use std::iter::repeat_with;
+
+pub struct GeneratePoStDynamicSectorsCountInput {
+    pub sector_bytes: PaddedBytesAmount,
+    pub challenge_seed: ChallengeSeed,
+    pub input_parts: Vec<(Option<String>, Commitment)>,
+}
+
+pub struct GeneratePoStFixedSectorsCountInput {
+    pub sector_bytes: PaddedBytesAmount,
+    pub challenge_seed: ChallengeSeed,
+    pub input_parts: [(Option<String>, Commitment); POST_SECTORS_COUNT],
+}
+
+pub struct VerifyPoStDynamicSectorsCountInput {
+    pub sector_bytes: PaddedBytesAmount,
+    pub comm_rs: Vec<Commitment>,
+    pub challenge_seed: ChallengeSeed,
+    pub proofs: Vec<[u8; 192]>,
+    pub faults: Vec<u64>,
+}
+
+pub struct VerifyPoStFixedSectorsCountInput {
+    pub sector_bytes: PaddedBytesAmount,
+    pub comm_rs: [Commitment; POST_SECTORS_COUNT],
```

Comment:
Similar as before, since there are common (unchanged) parameters here it's harder to grasp the difference between fixed and dynamic: only `comm_rs` and `faults` changing form `Vec` to fixed array.